### PR TITLE
Add ability to restore backup from configuration

### PR DESCRIPTION
**Pull request by pbrezina**
* New available commands `backup-list`, `backup-remove`, `backup-restore`.
* New available options `-b`, `-backup=NAME` for commands `select`, `apply-changes`, `enable-feature`, `disable-feature`.

There are quite a lot of commits but they are mostly small and standalone. So the review should not be that hard.

Resolves:
https://github.com/pbrezina/authselect/issues/120
https://github.com/pbrezina/authselect/issues/147

### CODE REVIEW

File: src/man/authselect.8.adoc [hunk 1]
```diff
@@ -52,13 +58,24 @@ To print help for the selected command run *authselect COMMAND --help*.
...
+*apply-changes* [-b] [--backup=NAME]::
     Re-apply currently selected profile. If the profile templates were updated
     this command can be used to regenerate current system configuration in
     order to apply these changes on the system. This command will only re-apply
     the changes if the existing configuration is a valid authselect
     configuration, otherwise an error is returned.
 
+    *-b*:::
+        Backup system files before applying changes. The backup
+        will be stored at {AUTHSELECT_BACKUP_DIR}/NAME. Current time with
```
Comment (thalman) [inline src/man/authselect.8.adoc]: Nitpicking: Should not be `stored as` instead of `stored at`?

Comment (pbrezina) [reply src/man/authselect.8.adoc]: Given `NAME` is a directory, I think using `at` is correct. But I am not a native speaker.

**Comment by pbrezina** [comment]
Thank you.

**Comment by pbrezina** [comment]
* `master`
  * 8f3d880991dfd73d63f58bd82452d0a8e913eabe - man: set keep threshold of transaltions to 70%
  * c2373fd9b5f20a353b393b53b5ff2303b27c25c2 - po: add CLI_TOOL_DELIMITER keyword
  * 3b1263c0a015142a0b3b947e6be00ebcbc949642 - po: update file list
  * 3e2e51df4d145daed2ed9b57ebec468cde2ddd52 - cli: support backup for apply-changes, enable-feature and disable-feature
  * e7a5778afed56c3f014175f7bbf680ba81b7a585 - cli: add backup-restore command
  * edd7226acdca01eff8cf4d16ce40d11c78119c4b - cli: add backup-remove command
  * 271ca635b3f69b3aaf1291061b0eeabe2cc5faeb - cli: add backup-list command
  * bd8b354682c423f550a1242c94920170a8e3a8fa - lib: sort backup by creation time
  * 097295bc22ca533afdf980d384ba5b46e749692d - lib: fix file_get_basename for path without directory
  * 39b593644ea5e7d592332d279bbcdad7389bdbd6 - lib: add restore ability
  * 63979c3b19d27c63c6d8dab629938408439c1b53 - lib: make selinux_get_default_context static
  * 8a4bdae9cde1e71fdc40f62919cf8e03139e1bf6 - lib: rename util/selinux.h guard due to conflict with selinux.h
  * 8a999ad5e583c9aac906a177fc8e0a441447a796 - lib: add selinux_copy_files_safely
  * 55e02139d0a7e93c6f6834ccc65593f8dbc03c48 - lib: add selinux_mkstemp_copy
  * cf713678bea6168316b14445ea3d57cf125b8c3f - lib: add file_mktmp_copy
  * d1f000fcf543c7b243bb1c093537244f4cb9cb12 - lib: rename textfile_copy to file_copy
  * 0add742394004ade7a5e4bdce279f13302add9d5 - lib: keep selinux context for backed up files
  * ea4a3bf7c31602e9c5147b3be53da9b15bb2c664 - lib: add selinux_file_copy
  * 91c92d77d57ce78d31ed3b6b17893514d6f098c7 - lib: add public functions to list and remove backups
  * d75921877d38131ee87b6df6400fc86165be63a3 - lib: add dir_remove to recursively remove directory
  * cbbea6b29e00661810726a1c0c620e5b0a0c36d8 - lib: put directory listing into its own utility module
  * 63a871303c053170e3a6776cea7c88fc411b134f - lib: backup also valid authselect configuration
  * 9eb34d0d90b94950cc3429e2eecbeb3fa3bc66b6 - lib: put config file name to standalone macro
  * dead5db3860a272e86d0980bd94b49cbdc9d6c32 - lib: refactor backup functionality
  * 73950ee94483a7bf6b2dc08a6fb430471685a739 - lib: use authselect file names for backup
  * 145ff9ef778ba929c4414e2ade97bd64f6e62937 - lib: fix typo in comment

**Pull request closed without merging**